### PR TITLE
Support test_non_ocs_taint_and_tolerations for external mode

### LINKED ISSUE #6995: Support test test_non_ocs_taint_and_tolerations for external mode

### PR DESCRIPTION
**Pull request by am-agrawa**
Fixes #6995

### CODE REVIEW

**Review by AviadP — DISMISSED** [review]

File: tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 1]
```diff
@@ -81,12 +81,18 @@ def test_non_ocs_taint_and_tolerations(self):
...
+        logger.info("Taint all nodes with non-ocs taint")
         ocs_nodes = get_worker_nodes()
         taint_nodes(nodes=ocs_nodes, taint_label="xyz=true:NoSchedule")
 
+        resource_name = constants.DEFAULT_CLUSTERNAME
+        if config.DEPLOYMENT["external_mode"]:
+            resource_name = constants.DEFAULT_CLUSTERNAME_EXTERNAL_MODE
+
         # Add tolerations to the storagecluster
+        logger.info("Add tolerations to storagecluster")
```
Comment (Shrivaibavi) [inline tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:93]: why are comment sessions added as log messages ? i see this change throughout the test

Comment (am-agrawa) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:93]: Bcz logs get printed during run time and helps debug if the test fails

Comment (Shrivaibavi) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:93]: Anybody who looks into the test directly or in this even the test logs understands the test execution with the current implementation. Hence lets remove the comment lines so we dont have to repeat the same statements in comment and in log messages.

Comment (am-agrawa) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:93]: sure, will remove the existing comments

**Comment by am-agrawa** [comment]
Test passed on 4.13 external mode deployment- http://pastebin.test.redhat.com/1103214

File: tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 2]
```diff
@@ -159,36 +168,37 @@ def test_non_ocs_taint_and_tolerations(self):
...
                 pod.delete(wait=False)
 
-        # After edit noticed few pod respins as expected
+        logger.info("After edit noticed few pod respins as expected")
         assert wait_for_pods_to_be_running(timeout=600, sleep=15)
 
-        # Check non ocs toleration on all pods under openshift-storage
+        logger.info("Check non-ocs toleration on all pods under openshift-storage")
         check_toleration_on_pods(toleration_key="xyz")
         self.sanity_helpers.health_check()
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:188]: For external mode we should use [external mode sanity helpers](https://github.com/red-hat-storage/ocs-ci/blob/b9b6fad319645f817d9b188cab7e7f53af0ec46c/ocs_ci/helpers/sanity_helpers.py#L178)

File: tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 3]
```diff
@@ -159,36 +168,37 @@ def test_non_ocs_taint_and_tolerations(self):
...
         number_of_pods_after = len(
             get_all_pods(namespace=config.ENV_DATA["cluster_namespace"])
         )
         assert (
             number_of_pods_before == number_of_pods_after
         ), "Number of pods didn't match"
 
-        # Add capacity to check if new osds has toleration
-        if not config.ENV_DATA["mcg_only_deployment"]:
+        logger.info("Add capacity to check if new osds has toleration")
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:198]: There is no need to log this message on the external mode cluster as we don't do it. Adding it here can create confusion. Please add this log information after the if condition.

Comment (Shrivaibavi) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:198]: +1

File: tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py (hunk 1 above)
Comment (am-agrawa) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:92]: Done

File: tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py (hunk 2 above)
Comment (am-agrawa) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py]: Done

File: tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py (hunk 3 above)
Comment (am-agrawa) [reply tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py]: Right, done

**Comment by am-agrawa** [comment]
The test is verified against the latest commit.

**Review by Shrivaibavi — APPROVED** [review]
LGTM

**Pull request merged**